lib/Service/FileService.php

```diff
@@ -1156,15 +1159,21 @@ public function addFile(ObjectEntity $objectEntity, string $fileName, string $co
                 schema: $objectEntity->getSchema()
             );
 
+            // Check if the content is base64 encoded and decode it if necessary
+            if (base64_encode(base64_decode($content, true)) === $content) {
+                $content = base64_decode($content);
+            }
```

Comment:
Although this looks valid, there is a case wherein the file can get garbled with this check, for example, a txt file containing only text without special characters (okay, wacky edge case, but still a valid file) will evaluate as valid base encoded data. Few examples:

```php
echo base64_encode(base64_decode('test', true));
```
returns `test`

```php
echo base64_encode(base64_decode('test Test', true));
```
returns `testTest`

```php
echo base64_encode(base64_decode('test Test 123', true));
```
returns testTest120=

We can decide to keep this check for now and fix this later, but then we should add a @todo